### ISSUE TITLE
Avoid hand-named files in favor of tmp files

### ISSUE THREAD

**Issue by kyleabeauchamp** [choderalab/gaff2xml#86]
We've had some issues with spaces in names.

1.  Use tempfiles everywhere to avoid cluttering workspace
2.  Ideally, we would like some way to still optionally report the paper trail of all intermediate files, as there are lots of moving parts at play

**Comment by davidlmobley** [comment]
I agree with this in general, though I've had some issues with old code I wrote which relied on temporary files everywhere - specifically, that when I run into unanticipated errors it can make it more difficult to troubleshoot (without going back and modifying the code) since I don't have/know where to find the temporary files. Obviously this can largely be handled by exception handling, so really the problem was more with my code than with temporary files itself.

However, I wonder if one way to simplify troubleshooting would be to use temporary directories rather than temporary files, so that when you encounter an exception you can point the user to the temporary directory containing all intermediate files (with some standardized names) used by the step which failed so they can look at what went wrong. Not a big difference, but perhaps slightly easier to work with.

Overall I agree though and we can work towards this.

**Comment by davidlmobley** [comment]
See also #82 which is another reason (I've run into before) to use temporary directories rather than temporary files.
